onnxruntime/contrib_ops/webgpu/bert/bias_split_gelu.cc

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/providers/webgpu/shader_helper.h"
+#include "core/providers/webgpu/webgpu_supported_types.h"
+#include "contrib_ops/webgpu/bert/bias_split_gelu.h"
+#include "contrib_ops/webgpu/bert/gelu.cc"
+#include "contrib_ops/webgpu/webgpu_contrib_kernels.h"
+
+namespace onnxruntime {
+namespace contrib {
+namespace webgpu {
+
+ONNX_OPERATOR_KERNEL_EX(
+    BiasSplitGelu,
+    kMSDomain,
+    1,
+    kWebGpuExecutionProvider,
+    (*KernelDefBuilder::Create())
+        .TypeConstraint("T", WebGpuSupportedFloatTypes()),
+    BiasSplitGelu);
+
+static int64_t GetMaxComponents(int64_t size) {
+  if (size % 4 == 0) {
+    return 4;
+  } else if (size % 2 == 0) {
+    return 2;
+  }
+  return 1;
+}
+
+Status BiasSplitGeluProgram::GenerateShaderCode(ShaderHelper& shader) const {
+  const ShaderVariableHelper& input = shader.AddInput("input");
+  const ShaderVariableHelper& bias = shader.AddInput("bias");
+  const ShaderVariableHelper& output = shader.AddOutput("output");
+
+  shader.AdditionalImplementation() << ErfImpl;
+
+  shader.MainFunctionBody() << shader.GuardAgainstOutOfBoundsWorkgroupSizes("uniforms.output_size")
+                            << "const M_SQRT2: f32 = sqrt(2.0);\n"
+                            << "const halfChannels = uniforms.channels / 2u;\n"
+                            << "let biasIdx = global_idx % halfChannels;\n"
+                            << "let batchIndex = global_idx / halfChannels;\n"
+                            << "let inputOffset = biasIdx + batchIndex * halfChannels * 2;\n"
+                            << "let valueLeft = " << input.GetByOffset("inputOffset") << " + " << bias.GetByOffset("biasIdx") << ";\n"
+                            << "let valueRight = " << input.GetByOffset("inputOffset + halfChannels") << " + " << bias.GetByOffset("biasIdx + halfChannels") << ";\n"
+                            << "let geluRight = valueRight * 0.5 * (erf_v(valueRight / M_SQRT2) + 1);\n"
+                            << output.SetByOffset("global_idx", "input_indices_t(valueLeft * geluRight)");
```

Comment:
should this be output_t?

Reply:
its actually unnecessary alltogether, thanks for catching